fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PruneProjectRule.java

```diff
@@ -50,12 +48,14 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
         if (((LogicalProjectOperator) input.getOp()).getColumnRefMap().isEmpty()) {
             Map<ColumnRefOperator, ScalarOperator> projectMap = Maps.newHashMap();
 
-            LogicalOperator logicalOperator = (LogicalOperator) input.inputAt(0).getOp();
+            OptExpression child = input.inputAt(0);
+            LogicalOperator logicalOperator = (LogicalOperator) child.getOp();
 
-            ColumnRefOperator smallestColumn = Utils.findSmallestColumnRef(
-                    logicalOperator.getOutputColumns(new ExpressionContext(input.inputAt(0))).getStream().
-                            mapToObj(context.getColumnRefFactory()::getColumnRef).collect(Collectors.toList()));
-            projectMap.put(smallestColumn, smallestColumn);
+            ColumnRefOperator smallestColumn =
```

Comment:
What does it mean by `smallestColumn`?

Reply:
For `Project Operator`, if its output columns are empty, we need to find a smallest column for it, since our plan node could not output empty.